background.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+// browser for Firefox, chrome for Chrome
+if (!browser) {
+  const browser = chrome;
+}
+
+// When the user clicks browserAction icon in toolbar, execute runLightbeam function
+browser.browserAction.onClicked.addListener(runLightbeam);
+
+async function runLightbeam() {
+
+	// Checks to see if Lightbeam is already open. Returns true if it is, false if not.
+	async function isOpen() {
+		const tabs = await browser.tabs.query({});
+		const fullUrl = browser.extension.getURL('index.html');
+		const lightbeamTabs = tabs.filter((tab) => {
+			return (tab.url === fullUrl);
+		});
+		return lightbeamTabs[0] || false;
+	}
+
+	// Only open a new Lightbeam instance if one isn't already open.
+	const lightbeamTab = await isOpen();
+	if (!lightbeamTab) {
+		browser.tabs.create(
+			{url: 'index.html'},
+			(tab) => {
+				const lightbeamTab = tab.id;
+				console.log(lightbeamTab);
+			}
+		);	
+	}
+}
```

Comment:
Also we will need to decide on a code style and agree on it. As this is yours and @princiya's project we should file a bug and get an ESLint config going. Personally my style is different but that is fine :)